### ISSUE TITLE
Set different optimization_options for instances of the same class

### ISSUE THREAD

**Issue by birgits** [oemof/oemof#170]
I created two instances of the storage class and want to set the optimization_options for storage1 to True and for storage2 to False. Is there a way to do that?

**Comment by gnn** [comment]
I guess you can just try to do `storage1.optimization_options = {'investment': True}`. If there is other stuff in `storage1.optimization_options` that you don't want to clobber, things get a bit more complicated but not by much. In that case you can try:

```python
import copy

storage1.optimization_options = copy.deepcopy(storage1.optimization_options)
storage1.optimization_options['investment'] = True
```

**Comment by birgits** [comment]
I quickly tried it, but it unfortunately didn't work... Since I needed a quick solution and the new solph allows for different optimization options for each instance anyway, I won't work on your suggested solution any further if that is okay with you. For now I just created a new class that inherits from the storage class so that I can set a different optimization option - quick and dirty ;)

**Comment by simnh** [comment]
Set general option for all storages to invest  True. The just set the maximum limit of additional capacity to zero for the ones that are fixed (no investment). That should work?

**Comment by birgits** [comment]
Generally yes, but in this case I want to set a time dependent ub_in for storage_2 (which I included in the set_bounds definition on my branch analogue to the time dependent ub_out). If the investment option is True the set_bounds definition is not called though. The charging limit is instead calculated using the c_rate. Now I probably could also have enabled a time dependent c_rate but since I already made the changes on ub_in I thought it would be easier to just set different optimization options... 

Thanks alot for you help @gnn and @simnh!
